manifest_test.go

```diff
@@ -470,15 +580,28 @@ func TestValidateProjectRoots(t *testing.T) {
 			stderrOutput.Reset()
 			err := ValidateProjectRoots(ctx, &c.manifest, sm)
 			if err != c.wantError {
-				t.Fatalf("Unexpected error while validating project roots:\n\t(GOT): %v\n\t(WNT): %v", err, c.wantError)
+				t.Fatalf("unexpected error while validating project roots:\n\t(GOT): %v\n\t(WNT): %v", err, c.wantError)
 			}
 
 			warnings := stderrOutput.String()
 			for _, warn := range c.wantWarn {
 				if !strings.Contains(warnings, warn) {
-					t.Fatalf("Expected ValidateProjectRoot errors to contain: %q", warn)
+					t.Fatalf("expected ValidateProjectRoot errors to contain: %q", warn)
 				}
 			}
 		})
 	}
 }
+
+func TestPruneOptionsFor(t *testing.T) {
+
+}
+
+func containsErr(s []error, e error) bool {
+	for _, a := range s {
+		if a.Error() == e.Error() {
```

Comment:
I think we can compare the error values directly and not the error message?

Reply:
Won't that compare pointers? Unless they refer to same pointer value they won't match.

Reply:
If we create variables for all the possible errors (which I think we almost do), the comparison would happen of the same pointer values. But, if we have some dynamic error, then this is fine.

Reply:
We still have some dynamic errors sadly.